android/java/org/chromium/chrome/browser/crypto_wallet/model/WalletListItemModel.java

```diff
@@ -50,11 +48,23 @@ public WalletListItemModel(
         mText2 = text2;
     }
 
-    public WalletListItemModel(int icon, String title, String subTitle, String text1, String text2,
-            boolean isImportedAccount) {
-        this(icon, title, subTitle, "", text1, text2);
-        mIsImportedAccount = isImportedAccount;
-        mIsAccount = true;
+    private WalletListItemModel() {}
+
+    public static WalletListItemModel MakeForAccountInfoWithBalances(
+            AccountInfo accountInfo, String fiatBalanceString, String cryptoBalanceString) {
+        WalletListItemModel result = new WalletListItemModel();
+        result.mIcon = Utils.getCoinIcon(accountInfo.accountId.coin);
+        result.mTitle = accountInfo.name;
+        result.mSubTitle = accountInfo.address;
+        result.mId = "";
+        result.mText1 = fiatBalanceString;
+        result.mText2 = cryptoBalanceString;
+        result.mAccountInfo = accountInfo;
+        return result;
+    }
+
+    public static WalletListItemModel MakeForAccountInfo(AccountInfo accountInfo) {
+        return MakeForAccountInfoWithBalances(accountInfo, null, null);
     }
```

Comment:
Added factory methods instead of obscure ctor for items having an associated account